src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -49,217 +47,294 @@ protected AbstractSensitivityAnalysis(MatrixFactory matrixFactory, Supplier<Grap
+
+        boolean isConnectedToComponent(Set<LfBus> connectedComponent);
+    }
+
+    abstract static class AbstractLfSensitivityFactor implements LfSensitivityFactor {
+
         // Wrap factors in specific class to have instant access to their branch and their equation term
         private final Object context;
 
-        private final String functionId;
-
         private final String variableId;
 
-        protected final LfBranch functionLfBranch;
+        protected final LfElement functionElement;
+
+        protected final SensitivityFunctionType functionType;
+
+        protected final SensitivityVariableType variableType;
 
         private Double predefinedResult = null;
 
         private double functionReference = 0d;
 
         private double baseCaseSensitivityValue = Double.NaN; // the sensitivity value on pre contingency network, that needs to be recomputed if the stack distribution change
 
-        private Status status = Status.VALID;
+        protected Status status = Status.VALID;
 
-        public LfSensitivityFactor(Object context, String functionId, String variableId, LfBranch functionLfBranch) {
+        public AbstractLfSensitivityFactor(Object context, String variableId,
+                                           LfElement functionElement, SensitivityFunctionType functionType,
+                                           SensitivityVariableType variableType) {
             this.context = context;
-            this.functionId = Objects.requireNonNull(functionId);
             this.variableId = Objects.requireNonNull(variableId);
-            this.functionLfBranch = functionLfBranch;
-            if (functionLfBranch == null) {
+            this.functionElement = functionElement;
+            this.functionType = functionType;
+            this.variableType = variableType;
+            if (functionElement == null) {
                 status = Status.ZERO;
             }
         }
 
+        @Override
         public Object getContext() {
             return context;
         }
 
-        public LfBranch getFunctionLfBranch() {
-            return functionLfBranch;
+        @Override
+        public String getVariableId() {
+            return variableId;
         }
 
-        public String getFunctionId() {
-            return functionId;
+        @Override
+        public SensitivityVariableType getVariableType() {
+            return variableType;
         }
 
-        public String getVariableId() {
-            return variableId;
+        @Override
+        public LfElement getFunctionElement() {
+            return functionElement;
+        }
+
+        @Override
+        public SensitivityFunctionType getFunctionType() {
+            return functionType;
         }
 
+        @Override
         public EquationTerm getEquationTerm() {
-            throw new NotImplementedException("getEquationTerm should have an override");
+            switch (functionType) {
+                case BRANCH_ACTIVE_POWER:
```

Comment:
Just a small remark here: maybe it could be better to use the terminology we already introduced: `BRANCH_P`, `BRANCH_I` (I have it on a working branch for transformer in current limiter mode) and `BUS_V`?

Reply:
I would keep it like this as it is supposed one day to be move to core as API V2 and as an officieal API is it more meaningful to keep  BRANCH_ACTIVE_POWER